coco/request_forwarder.py

```diff
@@ -54,6 +57,50 @@ def add_endpoint(self, name, endpoint):
         """
         self._endpoints[name] = endpoint
 
+    def start_prometheus_server(self, port):
+        """
+        Start prometheus server.
+
+        Parameters
+        ----------
+        port : int
+            Server port.
+        """
+        # Conect to redis
+        self.redis_conn = redis.Redis(host="localhost", port=6379, db=0)
+
+        def fetch_request_count():
+            for edpt in self._endpoints:
+                # Get current count and reset to 0
+                incr = int(self.redis_conn.getset(f"request_counter_{edpt}", "0"))
+                self.request_counter.labels(endpoint=edpt).inc(incr)
+
+        start_metrics_server(port, callbacks=[fetch_request_count])
+
+    def init_metrics(self):
+        """Initialise counters for every prometheus endpoint."""
+        # TODO: change description/name to dropped requests once that is in place
+        self.request_counter = Counter(
+            "coco_requests", "Count of requests received by coco.", ["endpoint"], unit="total"
+        )
+        self.result_counter = Counter(
+            "coco_results",
+            "Result of requests forwarded by coco.",
+            ["endpoint", "host", "port", "status"],
+            unit="total",
+        )
+        for edpt in self._endpoints:
+            for grp in self._groups:
+                for h in self._groups[grp]:
+                    self.request_counter.labels(endpoint=edpt).inc(0)
+                    hostname, port = h.hostname, h.port
+                    self.result_counter.labels(
```

Comment:
Does this rely on being initialized to 0 in any way? If so, it will probably not work, because there are more possible status codes than 0 and 200.

Reply:
I don't think it relies on it, my understanding was that until it is initialised it won't be retrievable. It's probably not a good idea to init all possible codes, but 0 and 200 are the only ones that are generated by the code at the moment.